clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/DeleteGvcfExecutor.scala

```diff
@@ -0,0 +1,157 @@
+            logger.info(s"Successfully deleted ${deleteResponses.size} Gvcfs.")
+            queryResponses
+          case _ =>
+            throw new Exception(
+              s"Deleted ${deleteResponses.size} Gvcfs. " +
+                s"Not all of the Gvcfs queried for were able to be deleted! Check the log for details"
+            )
+        }
+      }
+    }
+  }
+
+  private def deleteGvcfs(gvcf: Seq[TransferGvcfV1QueryOutput],
+                          webClient: ClioWebClient,
+                          ioUtil: IoUtil)(
+    implicit ec: ExecutionContext,
+    bearerToken: OAuth2BearerToken
+  ): Future[Seq[HttpResponse]] = {
+    if (gvcf.isEmpty) {
+      Future.failed(
+        new Exception(
+          s"No Gvcfs were found for ${deleteGvcf.transferGvcfV1Key}. Nothing has been deleted."
+        )
+      )
+    } else {
+      // Futures are transformed into Future[Either] so that any errors don't cause the entire resulting
+      // Future to be failed. Errors need to be preserved until all Futures have completed.
+      val deleteFutures = gvcf.map(
+        deleteGvcf(_, webClient, ioUtil)
+          .transformWith {
+            case Success(httpResponse) =>
+              Future(webClient.ensureOkResponse(httpResponse))
+            case Failure(ex) => Future.failed(ex)
+          }
+          .transformWith {
+            case Success(httpResponse) => Future.successful(Right(httpResponse))
+            case Failure(ex)           => Future.successful(Left(ex))
+          }
+      )
+      Future.foldLeft(deleteFutures)(Seq.empty[HttpResponse])((acc, cur) => {
+        cur match {
+          case Left(_)             => acc
+          case Right(httpResponse) => acc :+ httpResponse
+        }
+      })
+    }
+  }
+
+  private def deleteGvcf(gvcf: TransferGvcfV1QueryOutput,
+                         webClient: ClioWebClient,
+                         ioUtil: IoUtil)(
+    implicit ec: ExecutionContext,
+    bearerToken: OAuth2BearerToken
+  ): Future[HttpResponse] = {
+
+    def deleteInClio(): Future[HttpResponse] = {
+      logger.info(s"Deleting ${gvcf.prettyKey()} in Clio.")
+      webClient
+        .addGvcf(
+          TransferGvcfV1Key(
+            location = gvcf.location,
+            project = gvcf.project,
+            sampleAlias = gvcf.sampleAlias,
+            version = gvcf.version
+          ),
+          TransferGvcfV1Metadata(
+            documentStatus = Option(DocumentStatus.Deleted),
+            notes = gvcf.notes
+              .map(
+                notes => s"$notes\n${deleteGvcf.metadata.notes.getOrElse("")}"
+              )
+              .orElse(deleteGvcf.metadata.notes)
+          )
+        )
+        .map(webClient.ensureOkResponse)
+        .logErrorMsg(
+          s"Failed to delete the Gvcf ${gvcf.prettyKey()} in Clio. " +
+            s"The file has been deleted in the cloud. " +
+            s"Clio now has a 'dangling pointer' to ${gvcf.gvcfPath.getOrElse("")}. " +
```

Comment:
The expression `gvcf.gvcfPath.getOrElse("")` appears many times in this method, it should be stored in a `val`.